src/blocks/Block.as

```diff
@@ -1017,4 +1018,49 @@ public class Block extends Sprite {
 		return s.replace(/^/gm, "    ");
 	}
 
+	/* Blocking */
+
+	public function resetBlockingCount():void {
+		// should be called for a top stack block - resets all blocks in stack
+		resetArgsBlockingCount();
+		if(subStack1) subStack1.resetBlockingCount();
+		if(subStack2) subStack2.resetBlockingCount();
+		if(nextBlock) nextBlock.resetBlockingCount();
+	}
+
+	public function resetArgsBlockingCount():void {
+		// descend through all block's args, resetting to zero
+		blockingCount = 0;
+		var args:Array = this.args;
+		for(var i:uint=0; i<args.length; ++i) {
+			var barg:Block = args[i] as Block;
+			if(barg) barg.resetArgsBlockingCount();
+		}
+	}
+
+	public function increaseBlockingCount():void {
+		// increase blockingCount up the chain
+		// TLF: is it only the top level block that needs changing?
+		var b:Block = this;
+		while (true) {
+			b.blockingCount++;
+			if (b.parent is Block) b = Block(b.parent)
+			else return;
+		}
+		return; // never gets here
```

Comment:
Not a big deal, but this could be written more succinctly as:
```
while (b) {
	b.blockingCount++;
	b = b.parent as Block;
}
```
Also, the `return` at the bottom isn't necessary.